test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/AbstractCloudArchitectureIntegrationTest.java

```diff
@@ -0,0 +1,188 @@
+    }
+
+    private List<String> getAllContainersSpecIds(Collection<ContainerSpec> containerSpecs) {
+        List<String> ids = new ArrayList<>();
+        for (ContainerSpec containerSpec : containerSpecs) {
+            ids.add(containerSpec.getId());
+        }
+        return ids;
+    }
+
+    protected void verifyServerTemplateContainsKieServers(KieServerMgmtControllerClient kieControllerClient, String serverTemplate, int numberOfKieServers) {
+        Collection<ServerInstanceKey> kieServers = kieControllerClient.getServerTemplate(serverTemplate).getServerInstanceKeys();
+        assertThat(kieServers).hasSize(numberOfKieServers);
+    }
+
+    protected void verifySmartRouterContainsKieServers(KieServerRouterClient smartRouterAdminClient, int numberOfKieServers, List<String> containerIds, String... kieServerIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+
+        assertThat(routerConfig.getHostsPerServer()).containsKeys(kieServerIds);
+
+        for (String containerId : containerIds) {
+
+            // TODO: Commented until "Duplicate Kie server registration" issue gets resolved
+//            for (String kieServerId : kieServerIds) {
+//                assertThat(routerConfig.getHostsPerServer().get(kieServerId)).hasSize(numberOfKieServers);
+//            }
+            assertThat(routerConfig.getHostsPerContainer()).containsKey(containerId);
+            // TODO: Commented until "Duplicate Kie server registration" issue gets resolved
+//        assertThat(routerConfig.getHostsPerContainer().get(containerId)).hasSize(numberOfKieServers);
+            assertThat(routerConfig.getContainerInfosPerContainer()).containsKey(containerId);
+        }
+    }
+
+    protected void verifySmartRouterHostPerServerContainsContainers(KieServerRouterClient smartRouterAdminClient, String kieServerId, String... containersIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+        assertThat(routerConfig.getHostsPerServer().get(kieServerId)).contains(containersIds);
+    }
+
+    protected void verifySmartRouterHostPerContainerContainsServers(KieServerRouterClient smartRouterAdminClient, String containerId, String... kieServerIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+        assertThat(routerConfig.getHostsPerContainer().get(containerId)).contains(kieServerIds);
+    }
+
+    protected void scaleKieServerTo(int count, KieServerDeployment... deployments) {
+        scaleKieServerTo(count, Arrays.asList(deployments));
+    }
+
+    protected void scaleKieServerTo(int count, List<KieServerDeployment> deployments) {
+        for (Deployment deployment : deployments) {
+            deployment.scale(count);
+            deployment.waitForScale();
+        }
+    }
+
+    protected void assertActiveProcessInstanceCount(int expectedCount, QueryServicesClient... clients) {
```

Comment:
Again, `static`. Also, it's a method whose only purpose is to iterate over the argument. Why not have the method assert on the client directly, and iterate on the outside?

I can see some value here, but in order for that value to be realized, the method would have to use `SoftAssertions`.